src/chatd.cpp

```diff
@@ -5094,16 +5094,14 @@ bool Message::parseUrl(const std::string &url)
         }
     }
 
-    if (urlToParse.find("mega.co.nz/#!") != std::string::npos || urlToParse.find("mega.co.nz/#F!") != std::string::npos ||
-            urlToParse.find("mega.nz/#!") != std::string::npos || urlToParse.find("mega.nz/#F!") != std::string::npos ||
-            urlToParse.find("mega.nz/chat/") != std::string::npos)
+    std::regex megaUrlExpression("^((WWW.|www.)?mega.+(nz/|co.nz/)).*((#F!|#!|chat/)[a-z0-9A-Z-._~:\/?#!$&'()*+,;=\-@]+)$");
```

Comment:
Perhaps we should skip also the contact links: `https://mega.nz/C!<public_handle>`

Reply:
Apps will eventually show an enhanced UI for contact-links. Hence, the library should stop creating rich-links for that kind of links.
I'll fix it